source/extensions/filters/network/dubbo_proxy/router/router_impl.cc

```diff
@@ -331,7 +331,7 @@ void Router::UpstreamRequest::onPoolFailure(ConnectionPool::PoolFailureReason re
       reason == ConnectionPool::PoolFailureReason::RemoteConnectionFailure) {
     if (reason == ConnectionPool::PoolFailureReason::Timeout) {
       host->outlierDetector().putResult(Upstream::Outlier::Result::LocalOriginTimeout);
-    } else if (reason == ConnectionPool::PoolFailureReason::RemoteConnectionFailure) {
+    } else {
       host->outlierDetector().putResult(Upstream::Outlier::Result::LocalOriginConnectFailed);
     }
```

Comment:
Should we set a `LocalOriginConnectFailed` when the failture reason is `RemoteConnectionFailure`?

Reply:
In the outlier detection function, LocalOriginConnectFailed field is used to count connection errors，so i think we can set a LocalOriginConnectFailed when the failture reason is RemoteConnectionFailure or LocalConnectionFailure。

Reply:
[https://github.com/envoyproxy/envoy/blob/main/envoy/upstream/outlier_detection.h](url)